src/main/kotlin/it/pagopa/aca/services/AcaService.kt

```diff
@@ -1,16 +1,100 @@
+    @Autowired private val acaUtils: AcaUtils
+) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
     suspend fun handleDebitPosition(newDebtPositionRequestDto: NewDebtPositionRequestDto) {
-        // TODO add implementation here
-        logger.info("Handle debit position for amount ${newDebtPositionRequestDto.amount}")
+        logger.info(
+            "Handle debit position for amount: ${newDebtPositionRequestDto.amount} and iuv: ${newDebtPositionRequestDto.iuv}"
+        )
+        val requestId = UUID.randomUUID().toString()
+        val iupd = Iupd(newDebtPositionRequestDto.paFiscalCode, newDebtPositionRequestDto.iuv)
+        val paFiscalCode = newDebtPositionRequestDto.paFiscalCode
+        gpdClient
+            .getDebtPosition(paFiscalCode, iupd.value())
+            .filter { !acaUtils.isValidStatusForExecuteOperation(it.status) }
+            .switchIfEmpty(
+                Mono.error(
+                    RestApiException(
+                        HttpStatus.CONFLICT,
+                        "Unauthorized action",
+                        "Unauthorized action on debt position with iuv: ${newDebtPositionRequestDto.iuv}"
+                    )
+                )
+            )
+            .flatMap { debitPosition ->
+                if (acaUtils.checkInvalidateAmount(newDebtPositionRequestDto.amount)) {
+                    logger.info("Invalidate debit position with iupd: ${iupd.value()}")
+                    gpdClient.invalidateDebtPosition(paFiscalCode, iupd.value())
+                } else {
+                    logger.info("Update debit position with iupd: ${iupd.value()}")
+                    ibansClient
+                        .getIban(paFiscalCode, requestId)
+                        .map {
+                            acaUtils.updateOldDebitPositionObject(
+                                debitPosition,
+                                newDebtPositionRequestDto,
+                                iupd,
+                                it.second
+                            )
+                        }
+                        .flatMap { updatedDebitPosition ->
+                            gpdClient.updateDebtPosition(
+                                paFiscalCode,
+                                iupd.value(),
+                                updatedDebitPosition
+                            )
+                        }
+                }
+            }
+            .onErrorResume(GpdPositionNotFoundException::class.java) {
+                if (acaUtils.checkInvalidateAmount(newDebtPositionRequestDto.amount)) {
+                    logger.debug("Amount not compatible with the creation request")
+                    Mono.error(
+                        RestApiException(
+                            HttpStatus.BAD_REQUEST,
+                            "Unauthorized action",
+                            "Amount not compatible with the creation request"
```

Comment:
avoid details related a invalid and (maybe) unauthorized request

https://cheatsheetseries.owasp.org/cheatsheets/REST_Security_Cheat_Sheet.html#error-handling

Reply:
I had hypothesized this answer following the detail described in the SAMP 🤔 
Which answer do you think is more appropriate? @infantesimone

Reply:
If I understand correctly, this case is when a client tries to mark a debt position as closed for a non-existent position right (because `amount == 0`)?
In this case I would "downgrade" the response to a 404 HTTP status code (it's an operation on something that doesn't exist)

Otherwise, if passing information about the existence of the debt position is not OK I would use the same HTTP status code as in the other error cases (409 IIRC)

Reply:
Your idea about the 404 seems to me correct when it is an operation carried out on a resource never created.
It is also true, however, that it is a request formally wrong attributable to a 409 on the body of the request.
I propose a survey: 

**- 404 🍎**
**- 409 🍐**

@giovanniberti  @infantesimone  @ciuffagianluca  @pietro-tota  @MarcoAGreco

Reply:
What about use here a 422 instead? the request is formally valid (server can parse the request successfully) but instruction is not applicable

Reply:
Let's proceed then with the 422 👍🏻

Reply:
Fixed with: 05ae41bc417be4ef9eb3d169069fe8813b2e2e26